astropy/coordinates/matching.py

```diff
@@ -282,26 +283,20 @@ def search_around_3d(coords1, coords2, distlimit, storekdtree="kdtree_3d"):
 
     kdt1 = _get_cartesian_kdtree(coords1, storekdtree, forceunit=cunit)
 
-    # this is the *cartesian* 3D distance that corresponds to the given angle
-    d = distlimit.to_value(cunit)
-
+    all_matches = kdt1.query_ball_tree(kdt2, distlimit.to_value(cunit))
+    if not all_matches:
+        return _empty_match(coords1.distance.unit)
```

Comment:
Codecov is saying that this code isn't covered by tests. From what I can tell the lack of coverage is real, and more importantly this code branch cannot be reached. This means the functions can be simplified even more. I'll get to it tomorrow.

Reply:
This took a bit longer than I hoped it would because I discovered a bug in `search_around_3d()` that I had to fix first, and before I could fix the bug I had to refactor the tests so that I could write a sensible regression test for the bug.